### PR TITLE
[MRG+2] Removes OldScalarFormatter #7956 

### PR DESCRIPTION
**Pull request by NinadBhat**
#7956 
http://matplotlib.org/examples/pylab_examples/newscalarformatter_demo.html: drop OldScalarFormatter (which should probably be deprecated anyways...)

### CODE REVIEW

**Comment by NelleV** [comment]
I am sorry… Could you rename the file to plot_scalarformatter.py? We are currently migrating to sphinx-gallery, and it is possible that we will have to rename everything to plot_XX, so better renaming once than several times :)

**Comment by NinadBhat** [comment]
@NelleV should it be **plot_scalarformatter.py** or **plot_scalarformatter_demo.py**

**Comment by NelleV** [comment]
This looks good to me. I am going to make a couple of additional suggestions, but feel free to ignore those:

- Can you add a docstring and a title to this example, describing what the example shows? I can't say this is a great example, but here is an example of a docstring that describes what the example does: https://github.com/matplotlib/matplotlib/blob/master/examples/ticks_and_spines/plot_spines_bounds.py
- If you do add a docstring and a title, I think this example could fit in our "ticks and spines" category. Do not move it if you don't fullfill the first point, as this is going to break the migration to sphinx-gallery PR.

Thanks a lot for your contribution!

**Comment by NinadBhat** [comment]
@NelleV : I have tried to add a description let me know if this is ok or if futher improvment is needed.

2. I don't know what **"ticks and spines" category** is. Can you kindly explain?

Thanks

**Comment by NelleV** [comment]
@NinadBhat it's one of the category of our gallery. You can switch galleries by switching the folder in which the example lies. Right now, the example is in the folder examples/pylab_examples. To switch it to the ticks and spines category, just move the file in examples/ticks_and_spines.

File: examples/pylab_examples/plot_scalarformatter.py [hunk 1]
```diff
@@ -1,31 +1,21 @@
-import matplotlib.pyplot as plt
-import numpy as np
-from matplotlib.ticker import OldScalarFormatter, ScalarFormatter
+"""
+===========================
+Example For ScalerFromatter
```
Comment (NelleV) [inline examples/pylab_examples/plot_scalarformatter.py]: There is a small typo here: (ScalerFromatter -> ScalerFormatter).

File: examples/pylab_examples/plot_scalarformatter.py [hunk 2]
```diff
@@ -1,31 +1,21 @@
...
-fig.text(0.5, 0.975, 'The old formatter',
-         horizontalalignment='center', verticalalignment='top')
-ax1.plot(x * 1e5 + 1e10, x * 1e-10 + 1e-5)
-ax1.xaxis.set_major_formatter(OldScalarFormatter())
-ax1.yaxis.set_major_formatter(OldScalarFormatter())
-
-ax2.plot(x * 1e5, x * 1e-4)
-ax2.xaxis.set_major_formatter(OldScalarFormatter())
-ax2.yaxis.set_major_formatter(OldScalarFormatter())
+The example shows use of ScalerFormater with different settings.
```
Comment (NelleV) [inline examples/pylab_examples/plot_scalarformatter.py]: Another type here: ScalerFormater -> ScalerFormatter.

File: examples/pylab_examples/plot_scalarformatter.py (hunk 1 above)
Comment (NelleV) [inline examples/pylab_examples/plot_scalarformatter.py]: "Tick formatting using the ScalerFormatter" may be a better title?

Comment (QuLogic) [inline examples/pylab_examples/plot_scalarformatter.py]: And Scalar, not Scaler.

File: examples/pylab_examples/plot_scalarformatter.py (hunk 2 above)
Comment (dopplershift) [inline examples/pylab_examples/plot_scalarformatter.py]: Here too "Scalar"

**Comment by NinadBhat** [comment]
@NelleV  and @dopplershift sorry for the typos. I have made changes and moved the file to examples/ticks_and_spines.

File: examples/ticks_and_spines/plot_scalarformatter.py [hunk 3]
```diff
@@ -1,31 +1,21 @@
-import matplotlib.pyplot as plt
-import numpy as np
-from matplotlib.ticker import OldScalarFormatter, ScalarFormatter
+"""
+=========================================
+Tick formatting using the ScalarFromatter
```
Comment (NelleV) [inline examples/ticks_and_spines/plot_scalarformatter.py]: There is still a typo here (that I am fine fixing just before merging).

File: examples/tests/backend_driver.py [hunk 4]
```diff
@@ -205,7 +205,7 @@
     'multiline.py',
     'multiple_figs_demo.py',
     'nan_test.py',
-    'newscalarformatter_demo.py',
+    'plot_scalarformatter.py',
```
Comment (NinadBhat) [inline examples/tests/backend_driver.py]: @NelleV: This should go to **files['ticks_and_spines']**  right?

Comment (NelleV) [inline examples/tests/backend_driver.py]: If I understood properly from a conversation with @tacaswell, this file is way out of date and unmaintained.
@tacaswell can you confirm or correct this?

Comment (QuLogic) [inline examples/tests/backend_driver.py]: Maybe you're thinking of #8069?

Comment (NelleV) [inline examples/tests/backend_driver.py]: No I think of a later discussion (last week or so)

Comment (dstansby) [inline examples/tests/backend_driver.py]: Once #8069 goes through all these explicit example file references will be gone anyway, so I don't think this matters.

**Pull request merged**